### PR TITLE
Advanced calculator

### PR DESCRIPTION
**Pull request by vsupruniuk**

### CODE REVIEW

File: src/makeCalculator.js [hunk 1]
```diff
@@ -37,7 +37,39 @@
  * @return {object}
  */
 function makeCalculator() {
-  // write code here
+  const calculator = {
+    result: 0,
+
+    operate(oparation, number) {
```
Comment (illiarerun) [inline src/makeCalculator.js:43]:
```suggestion
    operate(operation, number) {
```

File: src/makeCalculator.js [hunk 2]
```diff
@@ -37,7 +37,39 @@
...
+
+    reset() {
+      calculator.result = 0;
+
+      return this;
+    },
+
+    add(number) {
+      calculator.result += number;
+    },
```
Comment (illiarerun) [inline src/makeCalculator.js:57]: right now we can execute this code:
image
i guess it should work a bit other way, we should change object that we called `operate` on, operation shouldn't be dependant on exact object, you can somehow pass needed object from operate to this actual `operation`

**Review by illiarerun — CHANGES_REQUESTED** [review]
GJ, now let's work on fixes:

File: src/makeCalculator.js [hunk 3]
```diff
@@ -37,7 +37,43 @@
...
+    result: 0,
+
+    operate(operation, number) {
+      this.result = operation(this.result, number);
+
+      return this;
+    },
+
+    reset() {
+      calculator.result = 0;
```
Comment (illiarerun) [inline src/makeCalculator.js:50]: use `this` here also, issue is still there

**Review by illiarerun — CHANGES_REQUESTED** [review]

**Review by jstmpelowycz — APPROVED** [review]

**Pull request closed without merging**